### PR TITLE
Added the utility Random.cs

### PR DESCRIPTION
**Pull request by IgnacioHdI**
Adds a Random static class that makes it easier for users to generate random numbers. It is a bit tedious to create random numbers, as you have to create a new Random variable, or use the RandomNumberGenerator. This additions makes it easier by adding several scripts to generate random ints, longs, doubles, bools as well as chars out of a string.

### CODE REVIEW

**Comment by AThousandShips** [comment]
Please open a [proposal](https://github.com/godotengine/godot-proposals) to gauge the need for this and allow discussion

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Random.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        public static float RandFloat(float minValue, float maxValue)
+        {
+            return random.NextSingle() * maxValue - minValue;
+        }
+        public static double RandDouble(double minValue, double maxValue)
+        {
+            return random.NextDouble() * maxValue - minValue;
+        }
+        public static short RandByte(short minValue, short maxValue)
```
Comment (AThousandShips) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Random.cs:24]:
```suggestion
        public static short RandShort(short minValue, short maxValue)
```

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Random.cs
Comment (AThousandShips) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Random.cs:1]: These methods need documentation comments

**Comment by raulsntos** [comment]
I don't think we should add this. There's already a way to get a shared instance of the `System.Random` class:

```csharp
int randomInt = Random.Shared.Next(0, 100);
```

There are also methods in `GD` to match what's available in the GDScript global scope:

```csharp
public static class GD
{
    public static float Randf();
    public static double Randfn(double mean, double deviation);
    public static uint Randi();
    public static double RandRange(double from, double to);
    public static int RandRange(int from, int to);
    public static uint RandFromSeed(ref ulong seed);
}
```

So it sounds like something that can be a library instead.

**Pull request closed without merging**